docs/versioning.rst

```diff
@@ -0,0 +1,156 @@
+**********
+Versioning
+**********
+
+
+Current Release Stages
```

Comment:
should we have a link in this section to the definitions below?

or link each status name in the table to the definition below? (footnote?)